components/sidebar_right_menu/index.js

```diff
@@ -15,15 +15,35 @@ import {isMobile} from 'utils/utils.jsx';
 import SidebarRightMenu from './sidebar_right_menu.jsx';
 
 function mapStateToProps(state) {
+    const license = state.entities.general.license;
     const config = getConfig(state);
     const rhsState = getRhsState(state);
 
     const enableTutorial = config.EnableTutorial === 'true';
     const tutorialStep = parseInt(get(state, Preferences.TUTORIAL_STEP, getCurrentUserId(state), TutorialSteps.FINISHED), 10);
 
+    const isLicensed = license.IsLicensed === 'true';
+    const appDownloadLink = config.AppDownloadLink;
+    const enableTeamCreation = config.EnableTeamCreation === 'true';
+    const enableUserCreation = config.EnableUserCreation === 'true';
+    const experimentalPrimaryTeam = config.ExperimentalPrimaryTeam;
+    const helpLink = config.HelpLink;
+    const reportAProblemLink = config.ReportAProblemLink;
+    const restrictTeamInvite = config.RestrictTeamInvite;
+    const siteName = config.SiteName;
```

Comment:
I may just be out-of-touch when it comes to JavaScript idioms, but it seems like defining these up here more than negates any benefit to using the key-value shorthand below. Is there a reason you don't just define them as key-value pairs below?

Reply:
I'd personally be a fan of inlining statements like these as long as their purpose is clear. It's really useful to have something complicated like `const isPostVisible = post.visibility == true && post.create_at > lastVisiblePostTime;`, but it feels like we're repeating ourselves with some of these. Only 1/5 though

Reply:
You're both absolutely right. There were a number of components where the prop computations were non-trivial and all factored out, so I began from that style and needlessly copied it throughout my PRs.

I'm hesitant to go back and change this with half of my PRs merged, but I'll make a note of this and investigate going back and fixing this all in one fell swoop.